### ISSUE TITLE
AU plugin won't output MIDI notes

### ISSUE THREAD

**Issue by emviveros** [pierreguillot/Camomile#271]
Using Camomile 1.0.8 beta4 and Camomile 1.0.7 with same behavior, Mac compiled AU plugin won't output MIDI notes, only bypass the MIDI notes. VST3 output the MIDI notes normally.

Test compiled plugins to test
[MAC_au_vst3_MIDIout.zip](https://github.com/pierreguillot/Camomile/files/6897648/MAC_au_vst3_MIDIout.zip)

I missed something?

Tested in Mac OSx 10.14.6, Ableton Live 10.

**Comment by pierreguillot** [comment]
Hello,

In fact, I'm not sure you can create an audio unit that accepts and generates audio and midi. That's might be a limitation. If you don't use audio, the plugin should try to use the midionly options. Perhaps the type of the plugin should also be modified, https://developer.apple.com/documentation/audiotoolbox/1584142-audio_unit_types.

**Comment by emviveros** [comment]
Thanks for response Pierre!

I tried to use midi only, but I can't do it work either in 1.0.7 and 1.0.8 Camomile in Ableton Live 10. All times I tried Live 10 crashs with AU, and return a fail message when tried to load the plugin in VST3 format similar to [this](https://github.com/pierreguillot/Camomile/issues/238#issue-799816619).

**Comment by pierreguillot** [comment]
> All times I tried Live 10 crashs with AU, 

Can you share the crash report and the new version of your plugin?

> return a fail message when tried to load the plugin in VST3 format similar to this.

Does it fix your issue to follow the answer?

**Comment by emviveros** [comment]
Error with configuration txt file with midionly true:
[midionly_true.zip](https://github.com/pierreguillot/Camomile/files/6930695/midionly_true.zip)
 - Camomile 1.0.7 - AU crash Live 10 - VST3 don't open
 - Camomile 1.0.8beta4 - AU crash Live 10 - VST3 return:
![Captura de Tela 2021-08-04 às 08 01 06](https://user-images.githubusercontent.com/3089536/128170316-ac646c82-3857-41b0-be0f-4260ed8e6c4f.png)

That's what you need? If I there's a way to do a more specific bug report, please tell me! 

All files, source and Camomile compilations are in the zip file.

**Comment by pierreguillot** [comment]
You should have crash reports https://support.apple.com/en-gb/guide/console/cnsl664be99a/mac

**Comment by emviveros** [comment]
Ok! Here's the crash report:
[crashReport_AUpluginCamomile1.0.8beta4.txt](https://github.com/pierreguillot/Camomile/files/6930952/crashReport_AUpluginCamomile1.0.8beta4.txt)

It happened trying open midionly true version of AU plugin compiled with Camomile 1.0.8beta4
> midionly_true.zip

**Comment by pierreguillot** [comment]
Indeed, it seems that MIDI only AudioUnit generates a crash in Live. The only thing I found about a MIDI only audio unit is the [kaudiounittype_midiprocessor](https://developer.apple.com/documentation/audiotoolbox/1584142-audio_unit_types/kaudiounittype_midiprocessor) but it's not explicitly that it should be MIDI only with no audio at all... I'm still investigating this issue.
